torch/csrc/jit/script/class_type.cpp

```diff
@@ -92,19 +58,56 @@ size_t ClassType::addAttribute(
   return slot;
 }
 
-const std::vector<Function*>& ClassType::methods() const {
-  return methods_;
+void ClassType::addMethod(Function* method) {
+  TORCH_CHECK(
+      getMethod(method->name()) == nullptr,
+      "Can't redefine method: ",
+      method->name(),
+      " on class: ",
+      python_str());
+  methods_.push_back(method);
 }
 
-ClassType::ClassType(
-    c10::optional<QualifiedName> name,
-    std::weak_ptr<CompilationUnit> cu,
-    bool is_module)
-    : NamedType(TypeKind::ClassType, std::move(name)),
-      compilation_unit_(std::move(cu)) {
-  if (is_module) {
-    parameterSlots_ = std::make_shared<std::vector<bool>>();
+Function* ClassType::getMethod(const std::string& name) const {
+  for (auto method : methods_) {
+    if (name == method->name()) {
+      return method;
+    }
+  }
+  return nullptr;
+}
+
+std::shared_ptr<CompilationUnit> ClassType::compilation_unit() {
+  auto cu = compilation_unit_.lock();
+  TORCH_INTERNAL_ASSERT(cu);
+  return cu;
+}
+std::shared_ptr<const CompilationUnit> ClassType::compilation_unit() const {
+  auto cu = compilation_unit_.lock();
+  TORCH_INTERNAL_ASSERT(cu);
+  return cu;
+}
+
+#ifndef USE_MOBILE_CLASSTYPE
```

Comment:
I don't see this set anywhere in this PR. How is this tested?

Reply:
The macro is currently for internal tests. Please refer to the internal diff D18226237 (which this PR is exported from) for testing cases. It's more urgent for internal experiment at this time. OSS tests and cmake changes will be added after the internal experiments are done.